mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Text, View, StyleSheet, Image, ScrollView } from 'react-native';
+import { useRoute } from '@react-navigation/native';
+import { MarkdownView } from 'react-native-markdown-view';
+import YoutubePlayer from 'react-native-youtube-iframe';
+import { BookScreenRouteProps } from '../navigation/HomeStackNavigator';
+import { LoadingCircle } from '../components/LoadingCircle';
+import { APIContext } from '../context/APIContext';
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
+  const { id, langs } = route.params;
+
+  const client = useContext(APIContext);
+  const i18nCtx = useContext(I18nContext);
+
+  // book screen states
+  const [loading, setLoading] = useState(true);
+  const [language, setLanguage] = useState(i18nCtx.locale.substring(0, 2));
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [image, setImage] = useState('');
+  const [tabBody, setTabBody] = useState('');
+  const [tabVideo, setTabVideo] = useState('');
+  const [activeButton, setActiveButton] = useState('');
+  const [tabContent, setTabContent] = useState(null);
```

Comment:
Any reason you split these all out into separate state variables instead of having a single `const [book, setBook] = useState()`? I think that would be a lot cleaner and (maybe?) more performant. React can combine state updates but it's not guaranteed, so we might get a lot of re-renders with separate vars.